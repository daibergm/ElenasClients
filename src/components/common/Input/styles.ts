import { StyleSheet } from 'react-native';

// @Assets
import { Colors } from '../../../assets/';

export default StyleSheet.create({
  container: {
    height: 48,
    borderBottomColor: Colors.secondary,
  },
  input: {
    fontSize: 16,
    color: Colors.primary,
  },
  inputError: {
    color: Colors.danger,
    fontSize: 12,
  },
});
