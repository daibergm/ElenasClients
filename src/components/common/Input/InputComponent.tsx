import React from 'react';
import { Input, InputProps } from 'react-native-elements';

// @Assets
import { Colors } from '../../../assets/';
import styles from './styles';

const InputComponent = ({ errorMessage, ...otherProps }: InputProps) => {
  return (
    <Input
      errorStyle={styles.inputError}
      placeholderTextColor={Colors.secondary}
      inputStyle={styles.input}
      inputContainerStyle={[
        styles.container,
        errorMessage ? styles.borderError : styles.borderSecondary,
      ]}
      errorMessage={errorMessage}
      {...otherProps}
    />
  );
};

export default InputComponent;
