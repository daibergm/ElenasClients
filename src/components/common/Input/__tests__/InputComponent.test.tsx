import React from 'react';
import { render } from '@testing-library/react-native';

// @Components
import Input from '../InputComponent';

// @Assets
import { Colors } from '../../../../assets/';

describe('Input test suits', () => {
  it('Render correctly', () => {
    const input = render(<Input />);
    expect(input).toMatchSnapshot();
  });
  it('Render without error', () => {
    const { getByTestId } = render(<Input testID="commonInput" />);
    const input = getByTestId('commonInput');
    expect(input.props.placeholderTextColor).toEqual(Colors.secondary);
  });
  it('Render with error', () => {
    const { getByTestId } = render(
      <Input testID="commonInput" errorMessage="Test error" />,
    );
    const input = getByTestId('commonInput');
    expect(input.props.placeholderTextColor).toEqual(Colors.danger);
  });
});
