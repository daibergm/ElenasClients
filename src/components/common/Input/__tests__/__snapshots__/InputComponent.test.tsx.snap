// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input test suits Render correctly 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    collapsable={false}
    nativeID="animatedComponent"
    style={
      Object {
        "alignItems": "center",
        "borderBottomColor": "rgba(0, 173, 238, 0.65)",
        "borderBottomWidth": 1,
        "borderColor": "#86939e",
        "flexDirection": "row",
        "height": 48,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
      }
    }
  >
    <TextInput
      editable={true}
      placeholderTextColor="rgba(0, 173, 238, 0.65)"
      style={
        Object {
          "color": "#00adee",
          "flex": 1,
          "fontSize": 16,
          "minHeight": 40,
        }
      }
      testID="RNE__Input__text-input"
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#dc3545",
        "fontSize": 12,
        "margin": 5,
      }
    }
  />
</View>
`;
